lightning/backend/localhelper.go

```diff
@@ -85,13 +87,31 @@ func (local *local) SplitAndScatterRegionByRanges(ctx context.Context, ranges []
 			continue
 		}
 
+		if len(regions) == 0 {
+			log.L().Warn("paginate scan region returns empty result", zap.Binary("minKey", minKey), zap.Binary("maxKey", maxKey),
+				zap.Int("retry", i))
+			return errors.New("paginate scan region returns empty result")
+		}
+
+		log.L().Info("paginate scan region finished", zap.Binary("minKey", minKey), zap.Binary("maxKey", maxKey),
+			zap.Int("regions", len(regions)))
+
 		regionMap := make(map[uint64]*split.RegionInfo)
 		for _, region := range regions {
 			regionMap[region.Region.GetId()] = region
 		}
 
 		var splitKeyMap map[uint64][][]byte
 		if len(retryKeys) > 0 {
+			firstKeyEnc := codec.EncodeBytes([]byte{}, retryKeys[0])
+			lastKeyEnc := codec.EncodeBytes([]byte{}, retryKeys[len(retryKeys)-1])
+			if bytes.Compare(firstKeyEnc, regions[0].Region.StartKey) < 0 || !beforeEnd(lastKeyEnc, regions[len(regions)-1].Region.EndKey) {
```

Comment:
Why compare firstKey and regions[0]?

Reply:
fistKey is the smallest key, so it should be included in region[0]; and lastKey in region[-1]

Reply:
It seems that region.StartKey without encode can not compare to other encoded key directly?

Reply:
Just like this:https://github.com/pingcap/tidb-lightning/blob/master/lightning/backend/local.go#L932